sdk/tables/azure-data-tables/azure/data/tables/aio/_table_client_async.py

```diff
@@ -0,0 +1,433 @@
+        try:
+            inserted_entity = await self._client.table.insert_entity(
+                table=self.table_name,
+                table_entity_properties=table_entity_properties,
+                **kwargs
+            )
+            properties = _convert_to_entity(inserted_entity)
+            return properties #Entity(properties)
+        except ValueError as error:
+            process_table_error(error)
+
+    @distributed_trace_async
+    async def update_entity(  # pylint:disable=R1710
+            self,
+            mode,  # type: Any
+            table_entity_properties,  # type: dict[str,str]
+            partition_key=None,  # type: Optional[str]
+            row_key=None,  # type: Optional[str]
+            etag=None,  # type: Optional[object]
+            match_condition=None,  # type: Optional[MatchCondition]
+            **kwargs  # type: Any
+    ):
+        # type: (...) -> None
+        """Update entity in a table.
+        :param mode: Merge or Replace entity
+        :type mode: ~azure.data.tables.UpdateMode
+        :param table_entity_properties: The properties for the table entity.
+        :type table_entity_properties: dict[str, str]
+        :param partition_key: The partition key of the entity.
+        :type partition_key: str
+        :param row_key: The row key of the entity.
+        :type row_key: str
+        :param etag: Etag of the entity
+        :type etag: str
+        :param match_condition: MatchCondition
+        :type match_condition: ~azure.core.MatchConditions
+        :return: None
+        :rtype: None
+        :raises: ~azure.core.exceptions.HttpResponseError
+        """
+
+        if_match, if_not_match = _get_match_headers(kwargs=dict(kwargs, etag=etag, match_condition=match_condition),
+                                                    etag_param='etag', match_param='match_condition')
+
+        if table_entity_properties:
+            partition_key = table_entity_properties['PartitionKey']
+            row_key = table_entity_properties['RowKey']
+            table_entity_properties = _add_entity_properties(table_entity_properties)
+
+        if mode is UpdateMode.REPLACE:
+            await self._client.table.update_entity(
+                table=self.table_name,
+                partition_key=partition_key,
+                row_key=row_key,
+                table_entity_properties=table_entity_properties,
+                if_match=if_match or if_not_match or "*",
+                **kwargs)
+        if mode is UpdateMode.MERGE:
+            await self._client.table.merge_entity(table=self.table_name, partition_key=partition_key,
+                                                  row_key=row_key, if_match=if_match or if_not_match or "*",
+                                                  table_entity_properties=table_entity_properties, **kwargs)
+
+    @distributed_trace
+    def query_entities(
+            self,
+            results_per_page=None,
+            select=None,
```

Comment:
there should be query_entities and list_entities: 

query_entities(self,filter,kwargs)  -> kwargs contains select, results_per_page

list_entities(self,kwargs) -> kwargs contains select, results_per_page, filter